### PR TITLE
Fix renderSubtreeIntoContainer to update context

### LINKED ISSUE #6599: context does not update using `unstable_renderSubtreeIntoContainer`
I use `ReactDOM.unstable_renderSubtreeIntoContainer` to render a portal.

```javascript
class Modal extends React.Component {

componentDidMount() {
    this.node = document.createElement('div');
    document.body.appendChild(this.node);
    this.renderPortal(this.props);
}

componentWillReceiveProps(nextProps) {
    this.renderPortal(nextProps);
}

componentWillUnmount() {
    ReactDOM.unmountComponentAtNode(this.node);
    document.body.removeChild(this.node);
}
    
renderPortal = (props) => {
    renderSubtreeIntoContainer(this, <ModalPortal {...props} />, this.node);
}

render() {
    return null;
  }
}
```

In my root component:

```javascript
class App extends React.Component {
    static childContextTypes = {
        highlightColor: React.PropTypes.object
    }

    getChildContext() {
        return {
            highlightColor: this.props.highlightColor
        }
    }

    ...
}
```

The context of root component will be updated and passed to children if  `this.props.highlightColor` is changed. This is right.
I can access the correct context in the `<ModalPortal/>` component in the first render.
And after that the context passed to  `<ModalPortal/>` **does not change any more** when the context of root component changed.
But I can access the correct context in the `<Modal/>`  every render.

what's wrong with my code ?

### PR DESCRIPTION
**Pull request by gaearon**
This includes a failing test case from #6924 (should now be passing) and should fix #6599.
Reviewers: @spicyj

### CODE REVIEW

File: src/renderers/dom/client/ReactMount.js [hunk 1]
```diff
@@ -439,6 +440,10 @@ var ReactMount = {
       null,
       nextElement
     );
+    var nextContext = parentComponent ?
+      parentComponent._reactInternalInstance._processChildContext(
```
Comment (gaearon) [inline src/renderers/dom/client/ReactMount.js]: Mr. Ugly but that’s exactly what we’ve been doing before (see below)

Comment (spicyj) [inline src/renderers/dom/client/ReactMount.js]: Can you change this to `ReactInstanceMap.get(parentComponent)`? Only that file should have `_reactInternalInstance` hardcoded.

File: src/renderers/shared/stack/reconciler/ReactUpdateQueue.js [hunk 2]
```diff
@@ -246,8 +246,9 @@ var ReactUpdateQueue = {
     enqueueUpdate(internalInstance);
   },
 
-  enqueueElementInternal: function(internalInstance, newElement) {
-    internalInstance._pendingElement = newElement;
+  enqueueElementInternal: function(internalInstance, nextElement, nextContext) {
+    internalInstance._pendingElement = nextElement;
+    internalInstance._context = nextContext;
```
Comment (spicyj) [inline src/renderers/shared/stack/reconciler/ReactUpdateQueue.js]: Technically we should probably queue context here too (`_pendingContext` or something) but it looks like the difference is probably unobservable right now? So I guess this is fine. Can you add a TODO here though?

**Comment by spicyj** [comment]
👍

**Pull request merged**